primitives/src/merkle_tree/internal.rs

```diff
@@ -226,6 +231,90 @@ where
     }
 }
 
+#[allow(clippy::type_complexity)]
+pub(crate) fn build_light_weight_tree_internal<E, H, I, Arity, T>(
+    height: usize,
+    elems: impl IntoIterator<Item = impl Borrow<E>>,
+) -> Result<(Box<MerkleNode<E, I, T>>, u64), PrimitivesError>
+where
+    E: Element,
+    H: DigestAlgorithm<E, I, T>,
+    I: Index + From<u64>,
+    Arity: Unsigned,
+    T: NodeValue,
+{
+    let leaves: Vec<_> = elems.into_iter().collect();
+    let num_leaves = leaves.len() as u64;
+    let capacity = BigUint::from(Arity::to_u64()).pow(height as u32);
```

Comment:
How large the tree can be? Is it necessary to use `BigUint`?

Reply:
Good point. For an append-only light-weight Merkle tree `u64` or even `u32` suffices. Let me change it, with an additional range check.

Reply:
Thanks, let's still use u64 for safety and future-compatibility.